homeassistant/components/xbox/const.py

```diff
@@ -0,0 +1,12 @@
+"""Constants for the xbox integration."""
+
+DOMAIN = "xbox"
+
+OAUTH2_AUTHORIZE = "https://login.live.com/oauth20_authorize.srf"
+OAUTH2_TOKEN = "https://login.live.com/oauth20_token.srf"
+
+HOME_LEGACY_PRODUCT_ID = "7b3ca835-5ef5-4d96-bc84-c1d8b5084236"
+
+APP_LEGACY_MAP = {
+    "Microsoft.Xbox.LiveTV_8wekyb3d8bbwe": "71e7df12-89e0-4dc7-a5ff-a182fc2df94f"
```

Comment:
What does the app legacy map and home legacy product id do?

Reply:
This is needed to get the proper app details from the catalog endpoint. Since “TV” and “Home” are built-in apps the “PackageFamilyName” method I use for everything else doesn’t work. These IDs don’t actually get returned from any previous API calls so I made it a map in case we found other apps that preform the same way.

Reply:
What's the reason this isn't part of the library? Is it the legacy status?

Reply:
No. The library simply wraps the catalog API. The legacy terminology is Microsoft’s.

Wanting to fetch the image of an App is part of the Home Assistant logic IMO. I didn’t want to add too much “business logic” to the API lib because not all use cases are th same.

99% of the time the value on the left (returned from the console API) can be passed directly into the catalog endpoint to get the app details. The console API doesn’t return the ID on the right, and returns 0 results if I use the one on the right.

It’s added purely to retrieve the app image and pretty media title. It could be removed if we want no image and a title derived off of the ID on the left.

Reply:
Why would a user of the library not want this workaround for that app_id?

Reply:
**Warning**: perhaps this is too deep for a PR review comment, but here it goes anyway...

I suppose it boils down to API design. The catalog endpoint is `/v7.0/products/lookup` and it has a set of well defined query string parameters (`alternateId` and `value` being two of them). The `xbox-webapi` exposes this through the `get_product_from_alternate_id` method which takes an ID value and ID type. The parent lib enumerates the known possible `alternateId` types which makes sense because this is a known enumerated field.

Now let's say we add logic to the API lib that says "for a given id and type combination, we know this won't return any results, so we will replace it with a different but equivalent id and type." This is the business logic I was referring to before.

Later down the road one of multiple scenarios can happen:
1) Microsoft updates their catalog database and adds additional IDs that can be linked, one of them being the ID/Type set that the user initially passed in.
2) Microsoft changes the ID that we are currently using to a different value.
3) Microsoft deprecates and removes the alternate id type that the library is replacing.

As a consumer of this library, I now have to wait for a library update and a version bump to fix any of the above issues. Microsoft's API structure/format didn't change, just the data it represents. We are now 1 or 2 years down the road and I as a maintainer of the `xbox-webapi` am getting pinged that something stopped working due to a business logic change and a library change is required to fix it. All projects that depend on that library have to wait for an update/bump.

Conversely, if the API library simply wraps the contract of the REST API without additional business logic, changes to data or responses can be handled and released by the developers of the project implementing the API. While it requires more intimate knowledge and debugging the underlying REST API to do so, it also helps reduce a problem Home Assistant consistently runs into: un-maintained dependancies breaking integrations. With this design, that only happens if the underlying API itself changes instead of the data it represents.

Reply:
It makes sense to follow the API spec in the library. On the other hand if the workaround would go in the library all library users would benefit. And change details would only need to be handled in the library instead of every app. As long as these details are universal for every app, it's not bad per se to put it in the library I think.

You know the API best, so you should decide. I think both ways work.

Reply:
So I think that the API wrapper should do just that, wrap the API. However, it could still contain the `APP_LEGACY_MAP` as an optional to use constant by consumers of the lib. We do that for example in [`aioshelly`](https://github.com/home-assistant-libs/aioshelly/blob/main/aioshelly/__init__.py#L10) for model names. They are not used in the API wrapper, but it's great not to expect each consumer of the lib to maintain their own.